test/integration/serviceaccount/service_account_test.go

```diff
@@ -247,16 +247,56 @@ func TestServiceAccountTokenAutoMount(t *testing.T) {
 	defaultMode := int32(0644)
 	expectedServiceAccount := serviceaccountadmission.DefaultServiceAccountName
 	expectedVolumes := append(protoPod.Spec.Volumes, v1.Volume{
```

Comment:
Use `createVolume` here instead, this is just maintenance burden and won't catch any failures

Reply:
This is a v1 type and is post defaulting. Admission works on the internal type pre-defaulting.

Reply:
> Admission works on the internal type pre-defaulting.

Not sure it matters for this discussion, but by the time an object gets to admission, it was read in external form, defaulted, and converted to internal form, so defaulting was already done

Reply:
The volume in question is an internal type constructed by the admission controller, not part of the request passed into admission, so not yet defaulted.

Reply:
In general, it's nuanced. Objects passed in via the request have undergone defaulting and conversion. In this specific instance, the internal volume type is created in the admission controller, not passed in from the request, so it has not undergone defaulting.